### ISSUE TITLE
Adding new error metric

### ISSUE THREAD

**Issue by namankumar** [automl/auto-sklearn#195]
I want to add new error metrics for regression (like RMSE and variants thereof) such that the model's cost function and ensemble selection by SMAC can be optimized for these metrics. How do I do this? 

Should I add the new metrics (lets say RMSE) to these files:
autosklearn/constants.py 
autosklearn/metrics/regression_metrics.py 
test/test_evaluation/evaluation_util.py 

The files in autosklearn/metalearning/files seem to relate to regression metrics but I don't understand their purpose. Please advise. Thanks!

**Comment by mfeurer** [comment]
PR #66 is about allowing custom metrics. Hopefully, I can merge it next week. Can you wait that long? I would like to not use the metrics subpackage any more, as I think the metrics in there should be replaced by the scikit-learn metrics in the long run to reduce the maintenance overhead. The files under `autosklearn/metalearning/files` are meta data for warmstarting as described in [our publication on auto-sklearn](https://papers.nips.cc/paper/5872-efficient-and-robust-automated-machine-learning.pdf)

**Comment by namankumar** [comment]
Yup, I'll wait for you to merge. Switching to scikit-learn's metrics is a really good idea, makes it easier for new users like me to get the hang of it faster. 

You guys have done some really excellent stuff here. That paper showed you used G-Means to cluster 140 datasets -- I also see auto-sklearn has code for G-Means and kNearestDatasets in autosklearn/metalearning/metalearning/. 

Is there more documentation on how to use GMeans and kNearestDataset with auto-sklearn? I would love to try them.

**Comment by namankumar** [comment]
Have you thought about how one could use sklearn's metrics with auto-sklearn?

**Comment by namankumar** [comment]
Hi there, wondering if you got the chance to merge PR #66 ?

**Comment by mfeurer** [comment]
No, sorry, I'm currently at the NIPS conference. Given my current schedule, I don't see this happening this year. Would you be willing to help on this? If yes, it would be helpful and speed up the process if you would rebase or merge away the current merge conflicts.

**Comment by namankumar** [comment]
For sure, happy to help. Enjoy NIPS. I will be there on Saturday as well.

**Comment by mfeurer** [comment]
Finally, auto-sklearn support custom metrics. Check out the example: http://automl.github.io/auto-sklearn/stable/manual.html#examples.
